### ISSUE TITLE
Help using generator-rise to create cli.

### ISSUE THREAD

**Issue by vproman** [bucaran/generator-rise#17]
Ability to generate cli using rise
OR
Add steps to documentation to manually create cli after generating module with rise.

I attempted to create cli starting with rise but npm link did not result in running cli bin name directly from command line.

**Comment by bucaran** [comment]
@vproman I like this idea. I'll work on something. :+1:

**Comment by Hypercubed** [comment]
Hey @bucaran, you might checkout out https://github.com/Hypercubed/generator-commander andf https://github.com/Hypercubed/autocmdr. They are both pretty old but could be usefull.

**Comment by bucaran** [comment]
@Hypercubed Thanks, the dilemma is the modules you are suggesting, while good in their own right, rely on commander.js and I would prefer this generator to be as _unopinionated_ as possible:

* `tape`, [TAP](testanything.org) producer, TAP itself has almost 30 years in the making
* `npm scripts` instead of the myriad of choices available
* `Babel` which for better or for worse is the best we've got if we want to run tomorrow's JavaScript

If possible, I want the simplest, most bare functionality and with as few deps as possible to aid in creating a CLI.

**Comment by Hypercubed** [comment]
@bucaran I agree.  Like I said I created these a long time ago.  If I were to do it over again I would handle many things differently.  That said, my idea for autocmdr is a set of modules that can be easily added to add CLI features to commander apps.  Some parts of this may be relevant if you make a CLI generator.

**Comment by bucaran** [comment]
@Hypercubed I'll definitely have a harder look. Woah, just noticed these two projects are almost 3 years old! :sweat_smile: Thanks!

**Comment by vproman** [comment]
I'm currently working with this pattern:
* bin folder for cli-specific scripts
* bin property in package.json defining mapping of command to cli-specific code in bin folder
* testing cli using npm link

I can run something like "my-command" without node or npm in the command and it executes the node cli code.  Using this to implement a team cli to automate common workflows.

**Comment by bucaran** [comment]
Hey @vproman!

It would be perfect if you could show a finished or work in progress project that we can use to get an idea of what you are looking for exactly and see how some of that can be applied to this generator.

**Comment by Hypercubed** [comment]
@bucaran Is you plan to add CLI features to this project or create a new project (`generator-rise-cli`)?

**Comment by bucaran** [comment]
@Hypercubed My plan is to add an extremely minimal set of features to help creating a CLI. Basically something like a `"main": "bin/index.js"` or `"main": "dist/index.js"` to `package.json`, and `chmod +x <that file>`.

**Comment by vproman** [comment]
I used this generator to get started: https://github.com/jonschlinkert/generator-node-cli
All a cli seems to require is a js file with hashbang to node and bin prop in package.json
```
#!/usr/bin/env node
console.log('my-cli.js');
```
```
  "bin": {
    "my": "./bin/my-cli.js"
  },
```
`npm link` will enable you to type `my` at command line and have it execute the js code.  Using `npm install -g my` will also enable the `my` command.
The main folder is still used to hold modules you might write specifically for the cli.

**Comment by vproman** [comment]
Also, a simple yes/no option of "do you want to create a cli" would be great.

**Comment by bucaran** [comment]
@vproman Here is the process to accomplish this:

1. Generate a new project with generator rise.
2. Add a `bin/` directory and `index.js`
```js
#!/usr/bin/env node
var my = require("../dist")
console.log( my() )
```
3. Add to your `package.json`
```json
  "bin": {
    "my": "bin/index.js"
  },
```
4. For the above to work, you also need to tweak `src/index.js`
```js
export function my () {
  return "hello, world!"
}
```
Note: You may use any name other than `index.js` if you wish.

**Comment by vproman** [comment]
I'd love the opportunity to contribute to this project!  Let me see what I can do.
